### PR TITLE
Fix temporary os download output option in device init

### PR DESCRIPTION
**Pull request by jviotti**

### CODE REVIEW

**Comment by abresas** [comment]
I suggest using https://github.com/raszi/node-tmp instead. We use it in other projects as well.

tmp.file() callback returns path, fd, and cleanup function that you can call in the end

```
tmp.file(function _tempFileCreated(err, path, fd, cleanupCallback) {
  if (err) throw err;

  console.log("File: ", path);
  console.log("Filedescriptor: ", fd);

  // If we don't need the file anymore we could manually call the cleanupCallback
  // But that is not necessary if we didn't pass the keep option because the library
  // will clean after itself.
  cleanupCallback();
});
```
It handles uncaught exceptions and other corner cases where you may have data leaks with

```
var tmp = require('tmp');
tmp.setGracefulCleanup();
```

Our images are huge files ~1GB so it's really sad if they are left on the user's device for no reason.

It uses `os.tmpDir` too internally.

**Comment by abresas** [comment]
The code comment I pasted from the project suggests that it's not necessary to clean. It will clean by itself when the process exits, but better do it early IMO.

**Comment by jviotti** [comment]
@abresas Implemented. Can you take another look please?

File: lib/actions/device.coffee [hunk 1]
```diff
@@ -210,10 +211,26 @@ exports.init =
 			(confirmed, callback) ->
 				return done() if not confirmed
 				options.yes = confirmed
-				osAction.download.action(params, options, callback)
 
-			(outputFile, callback) ->
+				# Cleanup the temporary files even when an uncaught exception occurs
+				tmp.setGracefulCleanup()
```
Comment (emirotin) [inline lib/actions/device.coffee]: maybe this should be done at the very beginning of the file

File: package.json [hunk 2]
```diff
@@ -66,6 +66,7 @@
     "resin-sdk": "^1.2.0",
     "resin-settings-client": "^1.0.0",
     "resin-vcs": "^1.2.0",
+    "tmp": "0.0.25",
```
Comment (emirotin) [inline package.json]: why using strict version here?

File: lib/actions/device.coffee (hunk 1 above)
Comment (abresas) [inline lib/actions/device.coffee]: Yeah as @emirotin says, it's a module-level setting so just after require

File: package.json (hunk 2 above)
Comment (abresas) [inline package.json]: I think npm install --save uses a strict version for some reason on tmp, I had it happen to me as well.
Maybe that's why it it happened here too. Better use ^

File: lib/actions/device.coffee (hunk 1 above)
Comment (jviotti) [inline lib/actions/device.coffee]: FIxed.

File: package.json (hunk 2 above)
Comment (jviotti) [inline package.json]: Fixed.

**Comment by abresas** [comment]
LGTM

**Pull request merged**